classes/PHPTAL.php

```diff
@@ -791,7 +847,31 @@ public function executeMacro($path)
     private function setCodeFile()
     {
         $this->findTemplate();
-        $this->_codeFile = $this->getPhpCodeDestination() . $this->getFunctionName() . '.' . $this->getPhpCodeExtension();
+        $this->_codeFile = $this->getPhpCodeDestination() . $this->getSubPath() . $this->getFunctionName() . '.' . $this->getPhpCodeExtension();
+    }
+
+    /**
+     * Generate a subpath structure
+     *
+     * @return string
+     */
+    private function getSubPath()
```

Comment:
Assuming that the feature is interesting (which is not up to me to decide), this sort of logic could be segregated into a separate class, so that instead of injecting various settings we inject an instance of the configured "writer" or "reader" (for example)